bindings/profilers/wall.cc

```diff
@@ -54,6 +54,29 @@ using namespace v8;
 
 namespace dd {
 
+int getTotalHitCount(const v8::CpuProfileNode* node) {
+  int count = node->GetHitCount();
+  auto child_count = node->GetChildrenCount();
+
+  for (int i = 0; i < child_count; ++i) {
+    count += getTotalHitCount(node->GetChild(i));
+  }
+  return count;
+}
+
+bool detectV8Bug(const v8::CpuProfile* profile) {
+  // Return true if either condition is false:
+  // - profile has at least one node with a non-zero hitcount
+  // - number of samples is strictly greater to total hitcount (meaning that
+  //   starting sample or some deopt events have been processed)
+  // If either condition is false, it implies that
+  // v8::SamplingEventsProcessor::ProcessOneSample loop is stuck for
+  // ticks_buffer_ or vm_ticks_buffer_.
+
+  auto totalHitCount = getTotalHitCount(profile->GetTopDownRoot());
+  return totalHitCount == 0 || profile->GetSamplesCount() == totalHitCount;
```

Comment:
Maybe just to be on the safe side?
```suggestion
  return totalHitCount == 0 || profile->GetSamplesCount() <= totalHitCount;
```

Reply:
I think it's better to check for equality, if sample count is strictly lower, it implies that tick samples were dropped and this is very likely a false positive.